substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ResourcesFeature.java

```diff
@@ -324,14 +324,15 @@ private static void scanJar(DebugContext debugContext, File root, Pattern[] incl
     }
 
     private static boolean matches(Pattern[] includePatterns, Pattern[] excludePatterns, String relativePath) {
+        String relativePathWithTrailingSlash = relativePath + "/";
```

Comment:
What if we have 2 `/`s?

Reply:
If we call this method from `scanDirectory`, the relative path is constructed here https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ResourcesFeature.java#L263-L268, and it will not have an extra slash at the end. If we call this method from `scanJar` the relative path is constructed here https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ResourcesFeature.java#L309-L309, and also there will be no extra slash. So in both variants, we are not accumulating slashes at the end.

Reply:
I think adding an assertion for that would be great maintenance-wise.  

Something like the following maybe:
```
assert Pattern.compile(".*\\w$").matcher(relativePath).matches();
```

Reply:
Sounds good. I'll add a check here.